src/c_api/c_predict_api.cc

```diff
@@ -243,6 +249,95 @@ int MXPredCreatePartialOut(const char* symbol_json_str,
   API_END_HANDLE_ERROR(delete ret);
 }
 
+int MXPredReshape(mx_uint num_input_nodes,
+                  const char** input_keys,
+                  const mx_uint* input_shape_indptr,
+                  const mx_uint* input_shape_data,
+                  PredictorHandle handle,
+                  PredictorHandle* out) {
+  MXAPIPredictor* p = static_cast<MXAPIPredictor*>(handle);
+  MXAPIPredictor* ret = new MXAPIPredictor();
+
```

Comment:
please wrap with unique_ptr and then if all is successful, assign with the result of .release()